client/v1/impl.go

```diff
@@ -114,8 +115,13 @@ func (c *httpClient) CreateRoot(
 	return &dir, nil
 }
 
-func (c *httpClient) ListRoots(ctx context.Context) (*v1.DirectoryList, error) {
-	resp, err := c.DoRaw(ctx, http.MethodGet, "/api/v1/roots", nil)
+func (c *httpClient) ListRoots(ctx context.Context, opts *storage.ListOptions) (*v1.DirectoryList, error) {
```

Comment:
You could turn this into a variadic function by changing the definition to:

```golang
func (c *httpClient) ListRoots(ctx context.Context, opts... storage.ListOptions) (*v1.DirectoryList, error) {
```

Note that I've omitted the usage of a pointer because we want to ensure that users calling this function pass valid references of `ListOptions`